paddle/phi/kernels/gpu/all_gather_gemm_kernel.cu

```diff
@@ -0,0 +1,442 @@
+
+#ifdef PADDLE_WITH_FLUX
+#include "paddle/phi/backends/dynload/flux.h"
+#endif
+#include "paddle/phi/core/distributed/comm_context_manager.h"
+
+#include "paddle/fluid/distributed/collective/process_group.h"
+#include "paddle/phi/backends/gpu/gpu_context.h"
+#include "paddle/phi/core/dense_tensor.h"
+#include "paddle/phi/kernels/empty_kernel.h"
+#include "paddle/phi/kernels/funcs/math_function.h"
+#include "paddle/phi/kernels/impl/slice_kernel_impl.h"
+#include "paddle/phi/kernels/slice_kernel.h"
+
+#include "paddle/phi/core/distributed/nccl_comm_context.h"
+#include "paddle/phi/core/distributed/utils.h"
+#include "paddle/phi/kernels/gpu/flux_utils.h"
+
+namespace phi {
+
+// Integration of Flux, a gemm-comm-overlap library as described in the paper
+// https://arxiv.org/pdf/2406.06858
+
+#ifdef PADDLE_WITH_FLUX
+template <typename InT, typename OutT>
+class AGGemmHelper {
+ public:
+  static constexpr int MAX_NUM_SIGNAL = 64;
+
+  using FlagType = int32_t;
+  const phi::GPUContext& dev_ctx;
+  distributed::NCCLCommContext* comm_ctx;
+  paddle::distributed::ProcessGroup* tp_group;
+  int32_t nnodes;
+  int32_t full_m;
+  int32_t n_dim;
+  int32_t k_dim;
+  const phi::DataType input_dtype =
+      std::is_same<InT, phi::dtype::float16>::value ? phi::DataType::FLOAT16
+                                                    : phi::DataType::BFLOAT16;
+  const phi::DataType output_dtype =
+      std::is_same<OutT, phi::dtype::float16>::value ? phi::DataType::FLOAT16
+                                                     : phi::DataType::BFLOAT16;
+  const bool transpose_weight;
+  const bool is_fp8_gemm{false};
+  int32_t rank;
+  int32_t world_size;
+  int32_t local_world_size;
+  int32_t local_rank;
+  // used for the cuda-ipc-barrier
+  std::vector<DenseTensor> sync_buffers;  // int32_t
+  std::vector<int32_t*> sync_buffer_ptrs;
+  std::vector<DenseTensor> input_buffers;    // InT
+  std::vector<DenseTensor> output_buffers;   // OutT
+  std::vector<DenseTensor> barrier_buffers;  // int32_t
+
+  std::vector<void*> input_buffer_ptrs;
+  std::vector<void*> output_buffer_ptrs;
+  std::vector<void*> barrier_buffer_ptrs;
+
+  DenseTensor input_buffer;
+  DenseTensor output_buffer;
+  DenseTensor barrier_buffer;
+
+  std::vector<void*> input_ptrs;
+  std::vector<FlagType*> barrier_ptrs;
+  AGRingMode ring_mode;
+  DenseTensor gemm_buffer;
+  size_t chunk_size;
+  size_t split_chunk_size;
+
+  int num_cp_streams;
+  std::vector<cudaStream_t> cp_streams;
+
+  cudaEvent_t cp_event;
+  cudaEvent_t ready_event;
+
+  AGGemmHelper(const phi::GPUContext& dev_ctx_,
+               paddle::distributed::ProcessGroup* tp_group_,
+               distributed::NCCLCommContext* comm_ctx_,
+               int32_t nnodes,
+               int32_t full_m,
+               int32_t n_dim,
+               int32_t k_dim,
+               bool transpose_weight = true,
+               AGRingMode ring_mode_ = AGRingMode::Auto)
+      : dev_ctx(dev_ctx_),
```

Comment:
In fact, according to Google coding style, the data member should be named like `dev_ctx_`, and the other member should be named like `dev_ctx`.